core/src/main/java/org/apache/iceberg/BaseEntriesTable.java

```diff
@@ -133,16 +133,13 @@ public CloseableIterable<StructLike> rows() {
       Types.NestedField readableMetricsField = projection.findField(MetricsUtil.READABLE_METRICS);
 
       if (readableMetricsField == null) {
-        CloseableIterable<StructLike> entryAsStruct =
-            CloseableIterable.transform(
-                entries(fileProjection),
-                entry -> (GenericManifestEntry<? extends ContentFile<?>>) entry);
-
         StructProjection structProjection = structProjection(projection);
-        return CloseableIterable.transform(entryAsStruct, structProjection::wrap);
+
+        return CloseableIterable.transform(
```

Comment:
I was trying to make both branches be more alike:

1. calculate final struct projection
2. calculate 'file' projection if needed for reading manifest
3. Use these to transform the result